spec_2/scrabble.js

```diff
@@ -0,0 +1,57 @@
+const playedTiles = "AERETOXMYCNS_B"
+
+const HIGHEST_TILE_COUNT = 12
+
+const tiles = {
+  A: { count: 9 },
+  B: { count: 2 },
+  C: { count: 2 },
+  D: { count: 4 },
+  E: { count: 12 },
+  F: { count: 2 },
+  G: { count: 3 },
+  H: { count: 2 },
+  I: { count: 9 },
+  J: { count: 1 },
+  K: { count: 1 },
+  L: { count: 4 },
+  M: { count: 2 },
+  N: { count: 6 },
+  O: { count: 8 },
+  P: { count: 2 },
+  Q: { count: 1 },
+  R: { count: 6 },
+  S: { count: 4 },
+  T: { count: 6 },
+  U: { count: 4 },
+  V: { count: 2 },
+  W: { count: 2 },
+  X: { count: 1 },
+  Y: { count: 2 },
+  Z: { count: 1 },
+  _: { count: 2 }
+}
+const tileFaces = [ 'A','B','C','D','E','F','G','H','I','J','K',
+  'L','M','N','O','P','Q','R','S','T','U','V','W', 'X','Y','Z','_' ]
+
+
+const reduceTileCounts = playedArray => {
+  playedArray.forEach( tile => tiles[ tile ].count-- )
+}
+
+const getNewTileCounts = () => {
+  for ( let x = HIGHEST_TILE_COUNT; x >= 0; x-- ) {
+    const tileFilter = tileFace => tiles[ tileFace ].count === x
+    tilesLeftInBag.push( `${x}: ${tileFaces.filter( tileFilter )}` )
+  }
+}
+
+const tilesLeftInBag = []
+
+const showRemainingTiles = ( tilesPlayed = playedTiles ) => {
+  reduceTileCounts( tilesPlayed.split('') )
+  getNewTileCounts()
```

Comment:
Encouragement for Object Oriented programming instead of declarative:
I'd expect getNewTileCounts() to take in an argument, rather than operating on external variables.